woocommerce/woocommerce-ios
# Mark order notification read if needed on watch app and notification extension

Closes: #13873


## Description


As a continuation on https://github.com/woocommerce/woocommerce-ios/pull/13734 this PR adds marking order notification as read on Watch app and in notification extension in a same logic as we do in the app, using the same endpoints but using `NotificationsRemote` directly instead of `NotificationAction` since `NotificationAction` is not available on the watch.

Updates for MarkOrderAsReadUseCase to have two versions of markOrderNoteAsReadIfNeeded, one usses `NotificationsRemote` and the second one uses `NotificationAction`. That way we can have same functionality in the iOS app, notification extension and watchOS app.

Good question would be why not just use `NotificationsRemote` in all places. When we use `NotificationAction` we can see that `NotificationStore` in `synchronizeNotification()` function also calls `updateLocalNotes(with: notes)` and that is a useful addition worth having 2 implementations.

## Testing information


General notification opening for the watchOS app and notification extension
- Do this testing on both watch app and notification extension. When testing the notification extension press the notification and check the notification preview is shown correctly.
- Get few new orders for one of the stores you have so it triggers notifications for it. It is important to have more of them so you can open order that is not the last.
- Tap on the notification for the order that was not the last order
- Check that the app opens the proper order
- Tap on the notification for the last order
- Check that the app opens the proper order and that the notification is marked as read also on the web

https://github.com/user-attachments/assets/80865274-ad87-4cee-a45b-ea1f198e3517

![IMG_723E898202D3-1](https://github.com/user-attachments/assets/f0185550-417e-4ddb-ba86-bda2249c8d01)

## Screenshots


There are no user-facing/UI changes but screenshots from https://github.com/woocommerce/woocommerce-ios/pull/13734 can be looked as a reference.

---

## Reviewer (or Author, in the case of optional code reviews):

Please make sure these conditions are met before approving the PR, or request changes if the PR needs improvement:
